### PR TITLE
Feat/strict cleanup

### PR DESCRIPTION
**Pull request by benschlueter**

### CODE REVIEW

**Comment by benschlueter** [comment]
[OpenSSL encryption test (use netcat to see if the connection is blocked)](https://help.mulesoft.com/s/article/How-to-set-up-a-minimal-SSL-TLS-server-from-the-command-line)

deploy in Constellation
`helm install cilium ./ --namespace kube-system --set endpointRoutes.enabled=true --set tunnel=disabled --set encryption.enabled=true --set encryption.type=wireguard --set encryption.nodeEncryption=true --set l7Proxy=false --set image.repository=ghcr.io/kakashiiiiy/cilium --set image.useDigest=false --set ipam.mode=kubernetes --set ipv4NativeRoutingCIDR=10.10.0.0/16  --set operator.image.repository=ghcr.io/kakashiiiiy/operator --set operator.image.suffix="" --set debug.enabled=true --set strictModeCIDRs="192.168.178.0/24 10.10.0.0/16"`

File: bpf/lib/wireguard.h [hunk 1]
```diff
@@ -76,6 +76,23 @@ is_encrypted(struct __ctx_buff *ctx)
...
     }
 
+    // workaround for cluster join constellation
+    if (bpf_htons(tcph->source) == 4000)
+    {
+        goto out;
+    }
+
+    // workaround for kubernetes join stuff
+    if (bpf_htons(tcph->dest) == 6443 || bpf_htons(tcph->source) == 6443)
```
Comment (3u13r) [inline bpf/lib/wireguard.h]: Shouldn't this be KubeAPI traffic which _should_ be TLS encrypted, is it not?

Comment (benschlueter) [reply bpf/lib/wireguard.h]: That is a good point. Somehow, a response to the new node is not encrypted when joining. Now sure what this is exactly, yet.

File: bpf/lib/wireguard.h [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    {
+        goto out;
+    }
+    // workaround for cilium status
+    if (bpf_htons(tcph->source) == 10250)
+    {
+        goto out;
+    }
+
+    // return false if the l7 layer only contains less than 4 bytes
```
Comment (3u13r) [inline bpf/lib/wireguard.h]:
```suggestion
    // return false if the l7 layer contains less than 4 bytes
```

File: bpf/lib/wireguard.h [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    if (length == 0)
+        goto out;
+
+    // workaround for cilium health check
+    if (bpf_htons(tcph->dest) == 4240 || bpf_htons(tcph->source) == 4240)
+    {
+        goto out;
+    }
+
+    // workaround for cluster join constellation
```
Comment (3u13r) [inline bpf/lib/wireguard.h]:
```suggestion
    // workaround for debugd traffic
```

**Review by 3u13r — CHANGES_REQUESTED** [review]

**Pull request closed without merging**